### PR TITLE
feat: add helperText API

### PR DESCRIPTION
**Pull request by DiegoCardoso**
- add helperText property to custom-field
- add styles for supported themes
- add theme variant for Lumo to set helperText above the field
- add demo for theme variant
- add unit and visual tests for helperText

Part of #72

### CODE REVIEW

File: src/vaadin-custom-field-mixin.html [hunk 1]
```diff
@@ -206,6 +223,21 @@
       }
     }
 
+    /** @private */
+    __setOrToggleHasHelperAttribute() {
+      const slottedNodes = this.shadowRoot.querySelector(`[name="helper"]`).assignedNodes();
+      // Only has slotted helper if not a text node
+      // Text nodes are added by the helperText prop and not the helper slot
+      // The filter is added due to shady DOM triggering this slotchange event on helperText prop change
```
Comment (yuriy-fix) [inline src/vaadin-custom-field-mixin.html:231]: Aligned with the comment above about the `slotchange` event 😅

File: test/custom-field-basic-test.html [hunk 2]
```diff
@@ -122,6 +130,71 @@
...
+        });
+
+        it('removing slotted helper removes has-helper attribute', function() {
+          const radioButtonGroupWithSlottedHelper = fixture('default-with-slotted-helper');
+
+          const helper = radioButtonGroupWithSlottedHelper.querySelector('[slot="helper"]');
+          radioButtonGroupWithSlottedHelper.removeChild(helper);
+          radioButtonGroupWithSlottedHelper.__helperObserver.flush();
+
+          expect(radioButtonGroupWithSlottedHelper.hasAttribute('has-helper')).to.be.false;
```
Comment (yuriy-fix) [inline test/custom-field-basic-test.html:195]: We should probably also test `aria-describedby` construction with `helperText` and `errorMessage`.

File: test/custom-field-basic-test.html [hunk 3]
```diff
@@ -122,6 +130,71 @@
...
+          helper.setAttribute('slot', 'helper');
+          helper.textContent = 'foo';
+          customField.appendChild(helper);
+          customField.__helperObserver.flush();
+
+          expect(customField.hasAttribute('has-helper')).to.be.true;
+        });
+
+        it('removing slotted helper removes has-helper attribute', function() {
+          const radioButtonGroupWithSlottedHelper = fixture('default-with-slotted-helper');
```
Comment (yuriy-fix) [inline test/custom-field-basic-test.html:189]: `radioButtonGroupWithSlottedHelper ` should be renamed 😅

File: test/visual/helper-text.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+<head lang="en">
+  <meta charset="UTF-8">
+  <title></title>
+  <script src="../../bower_components/webcomponentsjs/webcomponents-lite.js"></script>
+  <script>
+    const theme = window.location.search.replace(/.*theme=(\w+).*/, '$1') || 'lumo';
+    document.write(`
+      <link rel="import" href="../../theme/${theme}/vaadin-custom-field.html">
+      <link rel="import" href="../../../vaadin-text-field/theme/${theme}/vaadin-text-field.html">
+      <link rel="import" href="../../../vaadin-time-picker/theme/${theme}/vaadin-time-picker.html">
```
Comment (yuriy-fix) [inline test/visual/helper-text.html:12]: nit: `time-picker` is not used.

File: src/vaadin-custom-field-mixin.html [hunk 5]
```diff
@@ -96,6 +108,10 @@
         this.__setInputsFromSlot();
       });
 
+      this.__helperObserver = new Polymer.FlattenedNodesObserver(this.$.helperSlot, () => {
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply src/vaadin-custom-field-mixin.html]: Changed to `slotchange` to align with `vaadin-text-field`.

File: src/vaadin-custom-field-mixin.html [hunk 6]
```diff
@@ -69,7 +69,15 @@
         __errorId: String,
 
         /** @private */
-        __labelId: String
+        __labelId: String,
+
+        /** @private */
+        __helperTextId: {
+          type: String
+        },
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply src/vaadin-custom-field-mixin.html]: Done.

File: test/custom-field-basic-test.html (hunk 2 above)
Comment (DiegoCardoso) [reply test/custom-field-basic-test.html]: Done.

File: test/custom-field-basic-test.html (hunk 3 above)
Comment (DiegoCardoso) [reply test/custom-field-basic-test.html]: Removed.

File: test/visual/helper-text.html (hunk 4 above)
Comment (DiegoCardoso) [reply test/visual/helper-text.html]: Removed.

**Pull request merged**